tools/build.py

```diff
@@ -310,52 +346,102 @@ def Main():
               project_definition.name))
 
       if is_disabled:
-        disabled_packages.append(project_definition.name)
+        disabled_projects.append(project_definition.name)
       else:
         builds.append(project_definition)
 
   if not os.path.exists(options.build_directory):
     os.mkdir(options.build_directory)
 
+  undefined_projects = set(project_names)
+  for disabled_package in disabled_projects:
+    undefined_projects.remove(disabled_package)
+
+  check_configuration = set()
+  failed_builds = set()
+  failed_downloads = set()
+  missing_build_dependencies = set()
+
   current_working_directory = os.getcwd()
   os.chdir(options.build_directory)
 
-  undefined_packages = list(project_names)
-  for disabled_package in disabled_packages:
-    undefined_packages.remove(disabled_package)
+  try:
+    for project_definition in list(builds):
+      if project_names and project_definition.name not in project_names:
+        builds.remove(project_definition)
+        continue
 
-  failed_builds = []
-  for project_definition in builds:
-    if project_names and project_definition.name not in project_names:
-      continue
+      undefined_projects.remove(project_definition.name)
 
-    if undefined_packages:
-      project_index = undefined_packages.index(project_definition.name)
-      del undefined_packages[project_index]
+      if not project_builder.Download(project_definition):
```

Comment:
Please log "Downloading project: project_name" here, as for building (line 401)

Reply:
The title of the CL is "Split download, prebuild check and build phases". If these phases are meaningful enough to be a CL description, it would be helpful to know what phase a project build is in. 
I think these phases are a useful way to understand what build.py is doing, where problems are arising, and where to look for solutions, hence the request to log them.
As it was, I spent quite a while  on this CL trying to work out what these phases were, where they began and ended etc. This is likely to be replicated by users/reviewers in the future also.

Reply:
We discussed this, problems will primarily arise in the build phase, which are closely related to the build systems. Additional log statements for the phases do not add significant extra data for troubleshooting these.